plugin/storage/integration/integration_test.go

```diff
@@ -118,122 +127,128 @@ func (s *StorageIntegration) IntegrationTestGetOperations(t *testing.T) {
 		t.Log("\t Expected:", expected)
 		t.Log("\t Actual  :", actual)
 	}
-	assert.NoError(t, s.cleanUp())
 }
 
-func (s *StorageIntegration) IntegrationTestGetTrace(t *testing.T) {
-	t.Log("Testing GetTrace ...")
+func (s *StorageIntegration) testGetTrace(t *testing.T) {
+	defer s.cleanUp(t)
 
-	expected := s.getBasicFixture(t)
+	expected := s.loadParseAndWriteExampleTrace(t)
 	expectedTraceID := expected.Spans[0].TraceID
-	require.NoError(t, s.refresh())
+	s.refresh(t)
 
 	var actual *model.Trace
 	for i := 0; i < iterations; i++ {
 		s.logger.Info(fmt.Sprintf(waitForBackendComment, i+1, iterations))
 		var err error
-		actual, err = s.spanReader.GetTrace(expectedTraceID)
+		actual, err = s.SpanReader.GetTrace(expectedTraceID)
 		if err == nil && len(actual.Spans) == len(expected.Spans) {
 			break
 		}
 		time.Sleep(100 * time.Millisecond) // Will wait up to 3 seconds at worst.
 	}
 
 	CompareTraces(t, expected, actual)
-	assert.NoError(t, s.cleanUp())
 }
 
-func (s *StorageIntegration) IntegrationTestFindTraces(t *testing.T) {
-	t.Log("Testing FindTraces ...")
-	t.Log("\t(Note: all cases include ServiceName + StartTime range)")
-	queries, err := getQueries()
-	require.NoError(t, err)
-	for _, query := range queries {
-		t.Logf("\t\t* Query case: + %s", query.Caption)
-		s.integrationTestFindTracesByQuery(t, query.Query, query.ExpectedFixtures)
+func (s *StorageIntegration) testFindTraces(t *testing.T) {
+	defer s.CleanUp()
+
+	// Note: all cases include ServiceName + StartTime range
+	queryTestCases := loadAndParseQueryTestCases(t)
+
+	// Each query test case only specifies matching traces, but does not provide counterexamples.
+	// To improve coverage we get all possible traces and store all of them before running queries.
+	allTraceFixtures := make(map[string]*model.Trace)
+	expectedTracesPerTestCase := make([][]*model.Trace, 0, len(queryTestCases))
+	for _, queryTestCase := range queryTestCases {
+		var expected []*model.Trace
+		for _, traceFixture := range queryTestCase.ExpectedFixtures {
+			trace, ok := allTraceFixtures[traceFixture]
```

Comment:
not sure what this is used for, we don't seem to be updating the cache after creating a new trace

Reply:
it's used in L172 and then 174 and then 179

Reply:
sorry, I meant allTraceFixtures, doesn't seem like we set anything in it?